tests/tests_app/core/test_lightning_app.py

```diff
@@ -875,12 +878,44 @@ def test_slow_flow():
         )
     )
 
-    MultiProcessRuntime(app0).dispatch()
+    MultiProcessRuntime(app0, start_server=False).dispatch()
 
     app1 = LightningApp(
         SleepyFlowWithWork(
             sleep_interval=0.5 * FLOW_DURATION_THRESHOLD, work_sleep_interval=2 * FLOW_DURATION_THRESHOLD, parallel=True
         )
     )
 
-    MultiProcessRuntime(app1).dispatch()
+    MultiProcessRuntime(app1, start_server=False).dispatch()
+
+
+class SizeWork(LightningWork):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.counter = 0
+
+    def run(self, signal: int):
+        self.counter += 1
+
+
+class SizeFlow(LightningFlow):
+    def __init__(self):
+        super().__init__()
+        self.work0 = SizeWork(parallel=True, cache_calls=True)
+        self._state_sizes = {}
+
+    def run(self):
+        for idx in range(self.work0.counter + 2):
+            self.work0.run(idx)
+
+        self._state_sizes[self.work0.counter] = asizeof.asizeof(self.state)
+
+        if self.work0.counter >= 20:
+            self._exit()
+
+
+def test_state_size_constant_growth():
+    app = LightningApp(SizeFlow())
+    MultiProcessRuntime(app, start_server=False).dispatch()
+    assert app.root._state_sizes[0] == 5904
+    assert app.root._state_sizes[20] <= 8120
```

Comment:
The ideal test here will be to ensure the state size remains constant after thousands calls not only 20. We could mock the time.sleep() to make the test faster.

nit: it will be great to visualize it :D

Reply:
This is a slightly different problem.
Right now, the state stays constant assuming no other hash keys are added.